### ISSUE TITLE
RFC: minimal build container for zephyr

### ISSUE THREAD

**Issue by egernst** [zephyrproject-rtos/docker-image#30]
Glad to see there's a containerized environment to support building zephyr.

I have a more narrowly focused use case that I'd like supported, and am looking for feedback:

## Background:
I'd like a container that I can use for building the sources both as part of my development flow, as well as through CI testing.

## Issue with current implementation:
The container image supports VNC, which is helpful for some, but is excessive for my use case. Aside from size addition, it seems to have necessitated creating users, etc.

These users don't necessarily map to who will ultimately use the container to build and then utilze the built artifacts to test/run.

I'd like to be able to execute a build as follows:
docker run -it --user "$(id -u):$(id -g)" -v /my/west/workdir /home/eric/src/zephyrproject:/workdir --workdir /workdir/zephyr  bash -c "west build with my args" 

This allows me to obtain artifacts that are owned by the calling user.

## Query

Does it make sense to offer a build environment that doesn't include VNC and container-build-time defined users?

**Comment by egernst** [comment]
:wave: @nashif - let me know what you think...

**Comment by nashif** [comment]
@egernst  yes, it makes sense, do you want to submit a simplified version?

**Comment by joectchan** [comment]
I used the current docker image to build USB DFU sample application. One of the steps was signing the binary. However, the current image does not include  python package cryptography. I need to `pip3 install cryptography` manually. I guess a docker image to build real production code usually would want to sign the binary. I recommend to include cryptography.

I sometimes wish to use less and vi to fix minor mistake inside the context of docker. That is straightly for convenience. Please consider including these small footprint high impact tools.

**Comment by MeisterBob** [comment]
I made myself a [development container](https://github.com/MeisterBob/zephyr-docker) that is 4.23GB with Zephyr and the arm toolchain already included. To achieve this, I removed tensorflow and the biggest hals that I don't need. Maybe it helps someone.
